parties/opkclient.go

```diff
@@ -23,49 +25,77 @@ type MFACos interface {
 }
 
 type OpkClient struct {
-	PktCom      []byte
-	Signer      *pktoken.Signer
+	Pkt         *pktoken.PKToken
 	Op          OpenIdProvider
 	MFACosigner MFACos
 }
 
-func (o *OpkClient) OidcAuth() ([]byte, error) {
-	nonce, err := o.Signer.GetNonce()
+func (o *OpkClient) OidcAuth(
+	signer crypto.Signer,
+	alg jwa.KeyAlgorithm,
+	extraClaims map[string]any,
+	signGQ bool,
+) ([]byte, error) {
+	// Use our signing key to generate a JWK key with the alg header set
+	jwkKey, err := jwk.PublicKeyOf(signer)
+	if err != nil {
+		return nil, err
+	}
+	jwkKey.Set(jwk.AlgorithmKey, alg)
```

Comment:
Do we need to check that `alg` matches the `signer`? Or can we infer `alg` from `signer`? I'm wondering what would happen if I provided an ECDSA key with `alg` set to "RS256".

Reply:
I did look into this because I agree that would just be incredibly nice, but I couldn't figure out any way to do so. We could check that the `alg` is of the same type as the `signer`, but even then we'd probably do so by type checking against the golang standard library which would probably be overly restrictive for anyone using a different library that implements `crypto.Signer`.

The error message I got from trying to sign with `alg=ES256` using an RSA key was:
```
failed to generate signature for signer #0 (alg=ES256): failed to sign payload: failed to unmarshal ASN1 encoded signature: asn1: structure error: length too large
```
So....not very helpful. Probably partly because the `lestrrat-go/jwx` library doesn't type-check its keys.

I actually went back and forth on this in my implementation and here's my logic:
It's hard to be helpful here because either we can have a coupled interface where we're in charge of generating, storing, and parsing the key and alg information which makes us much less flexible against new or different signing protocols, OR we leave this as uncoupled interfaces and there is some definite room for user error that will be frustrating to debug but we are more flexible. I thought flexibility was important enough to go with the latter.

Reply:
@EthanHeilman